test/redux/StateTransitionBehvaior.js

```diff
@@ -0,0 +1,223 @@
+import { handleTaskflowChange } from '../../src/StateTransitionBehavior';
+
+import * as ProjectActions  from '../../src/redux/actions/projects';
+import * as TaskflowActions from '../../src/redux/actions/taskflows';
+import * as ClusterActions  from '../../src/redux/actions/clusters';
+import * as FSActions       from '../../src/redux/actions/fs';
+
+// a full redux state, mostly empty, 1 project with two simulations
+import fullState from '../sampleData/basicFullState';
+import taskflowState from '../sampleData/basicTaskflowState';
+
+import expect from 'expect';
+import { registerAssertions } from 'redux-actions-assertions/expect';
+import deepClone    from 'mout/src/lang/deepClone';
+
+/* global describe it beforeAll afterEach afterAll*/
+registerAssertions();
+
+// we spy on redux actions here, and they just need to return an action with some type
+const emptyAction = { type: 'NO-OP' };
+function setSpy(target, method, data) {
+  expect.spyOn(target, method)
+    .andReturn(data);
+}
+
+describe('StateTransitionBehavior', () => {
+  const taskflowId = '574c9d900640fd6e133b4b57';
+  const taskId = '574c9f350640fd6e13b11e39';
+  fullState.taskflows = deepClone(taskflowState);
+
+  it('should do nothing if there\'s no taskflow', () => {
+    expect(handleTaskflowChange({})).toBe(undefined);
+  });
+
+  describe('simulation status and taskflow actions', () => {
+    const taskflow = deepClone(fullState.taskflows.mapById[taskflowId]);
+    const simulation = deepClone(fullState.simulations.mapById['574c8aa00640fd3f1a3b379f']);
+    const metadata = Object.assign({}, simulation.metadata, { status: 'complete' });
+    const newMeta = {
+      actions: ['rerun'],
+      allComplete: false,
+      outputDirectory: undefined,
+      primaryJob: 'pyfr_run',
+    };
+
+    beforeAll(() => {
+      setSpy(ProjectActions, 'saveSimulation', emptyAction);
+      setSpy(TaskflowActions, 'updateTaskflowMetadata', emptyAction);
+    });
+
+    afterAll(() => {
+      expect.restoreSpies();
+      fullState.preferences.clusters.mapById = {};
+    });
+
+    it('should set status to terminated, rerun in actions', () => {
+      // if there is 1 terminated job, the status is terminated
+      taskflow.jobMapById = { someId: { _id: 'someId', status: 'terminated' } };
+      taskflow.allComplete = false;
+      metadata.status = 'terminated';
+      handleTaskflowChange(fullState, taskflow);
+      expect(ProjectActions.saveSimulation).toHaveBeenCalledWith(Object.assign({}, simulation, { metadata }));
+      expect(TaskflowActions.updateTaskflowMetadata).toHaveBeenCalledWith(taskflowId, newMeta); // actions: ['rerun']
+    });
+
+    it('should set status to terminated, rerun in actions', () => {
+      // if there is 1 terminated job, the status is terminated
+      taskflow.jobMapById = {};
```

Comment:
I was referring to the comment, wasn't sure why it was the same as the test above?